Automattic/jetpack
# Backups: Convert to feature checks

#### Changes proposed in this Pull Request:

* No functional changes, everything should work as before
* Moves from plan checks to feature checks to determine backups support.

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Set up a jurassic ninja site with Jetpack beta and select this branch.
* Upgrade to a daily backup plan and save the credentials.
* Verify the Backups box in At a Glance looks like this: 
<img width="443" alt="image" src="https://user-images.githubusercontent.com/1398304/165589309-420d9a2d-9bb1-4ff2-bdda-be82ebd92983.png">

* Upgrade to a real time backup plan (like Complete)
* Verify the Backups box in At a Glance looks like this: 
<img width="441" alt="image" src="https://user-images.githubusercontent.com/1398304/165589621-71f9edb3-a3cb-47c3-a9e1-06849d135e29.png">
